// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTServerContext";
option optimize_for = LITE_RUNTIME;

import public "ConsensusEnums.proto";
import public "PendingCommand.proto";

message ServerContext {
    optional uint32 version = 1;
    optional string serverid = 2;
    optional uint64 highesttransactionnumber = 3;
    repeated uint64 tentativerequestnumber = 4;
    optional uint64 revision = 5;
    optional string adminpassword = 6;
    optional bool adminattempted = 7;
    optional bool adminsuccess = 8;
    optional DeliveryState state = 9;
    optional LastReplyStatus laststatus = 10;
    optional PendingCommand pending = 11;
}
